FestivalPlanner/screenings/views.py

```diff
@@ -990,14 +997,20 @@ def _get_ticket_warning_choices(warning_type, fan, screening):
         return choices
 
     def _get_choices_with_link(self, warning_type, fan, screening):
+        session = self.request.session
         fix_wording = ScreeningWarning.fix_by_warning[warning_type]
 
         # Set up the queryset to select screenings with the given fan and warning.
         filters = [self.reset_filter, self.filter_by_fan[fan], self.filter_by_warning_type[warning_type]]
         if warning_type == ScreeningWarning.WarningType.ATTENDS_SAME_FILM:
             filters.append(self.filmscreenings_filter)
         link_wording = ScreeningWarning.link_wording_by_ticket_warning[warning_type]
-        querystring = Filter.get_display_query_from_keys([f.get_cookie_key() for f in filters], on=True)
+        link_active = self.filter_by_fan[fan].on(session) and self.filter_by_warning_type[warning_type].on(session)
+        if link_active:
+            link_url = None
+        else:
+            querystring = Filter.get_display_query_from_keys([f.get_cookie_key() for f in filters], on=True)
+            link_url = '/screenings/warnings' + querystring
```

Comment:
When executing a query that works on the same page, the slug isn't necessary.
Please use the query string as `link_url`.